docs/docsite/rst/dev_guide/developing_locally.rst

```diff
@@ -17,89 +17,97 @@ Extending Ansible with local modules and plugins offers lots of shortcuts such a
+Adding standalone local modules for selected playbooks or a single role
+-----------------------------------------------------------------------
+
+Ansible automatically loads all executable files from certain directories adjacent to your playbook or role as modules. Standalone modules in these locations are available only to the specific playbook, playbooks, or role in the parent directory.
 
-* In a selected playbook or playbooks: Store the module in a subdirectory called ``library`` in the directory that contains those playbooks.
-* In a single role: Store the module in a subdirectory called ``library`` within that role.
+* To use a standalone module only in a selected playbook or playbooks, store the module in a subdirectory called ``library`` in the directory that contains the playbook or playbooks.
+* To use a standalone module only in a single role, store the module in a subdirectory called ``library`` within that role.
 
 .. _distributing_plugins:
 .. _local_plugins:
 
-Adding a plugin locally
-=======================
-Ansible loads plugins automatically too, and loads each type of plugin separately from a directory named for the type of plugin. Here's the full list of plugin directory names:
-
-    * action_plugins*
-    * cache_plugins
-    * callback_plugins
-    * connection_plugins
-    * filter_plugins*
-    * inventory_plugins
-    * lookup_plugins
-    * shell_plugins
-    * strategy_plugins
-    * test_plugins*
-    * vars_plugins
+Adding a plugin locally outside of a collection
+===============================================
 
-.. note::
+You can configure Ansible to load standalone local plugins in a specified location or locations and make them available to all playbooks and roles. Alternatively, you can make a standalone local plugin available only to specific playbooks or roles.
+
+Adding local plugins for all playbooks and roles
+-------------------------------------------------
+
+To load standalone local plugins automatically and make them available to all playbooks and roles, use the configuration setting or environment variable for the type of plugin you are adding. These configuration settings and environment variables take colon-separated list, similar to ``$PATH``, and default to ``~/.ansible/plugins/<plugin_type>/:/usr/share/ansible/plugins/<plugin_type>/``. You have two options:
 
-	After you add the plugins and verify that they are available for use, you can see the documentation for all the plugins except for the ones marked with an asterisk (*) above.
+* Add your local plugin to one of the default configured locations:
+   * the directory named for the correct ``plugin_type`` within ``~/.ansible/plugins/`` - for example, ``~/.ansible/plugins/callback/``
+   * the directory named for the correct ``plugin_type`` within ``/usr/share/ansible/plugins/`` - for example, ``/usr/share/ansible/plugins/action/``
```

Comment:
same here, these directories are not guaranteed

Reply:
I suggest pointing people at `ansbile-config dump |grep plugin_type_PATH` (not all use DEFAULT_, specifically become and doc fragment plugins)

Reply:
addressed in 22e6bc666ecca573e9a5a86acd2ac64cec01a223

Reply:
and also in c3790e3887d1eb3d7b3437f5ec4794233f1ff03d